libs/vm-modules/include/vm_modules/ml/model/model.hpp

```diff
@@ -66,20 +83,26 @@ class VMModel : public fetch::vm::Object
       fetch::vm::VM *vm, fetch::vm::TypeId type_id,
       fetch::vm::Ptr<fetch::vm::String> const &model_category);
 
-  void LayerAddDense(fetch::vm::Ptr<fetch::vm::String> const &layer, math::SizeType const &inputs,
-                     math::SizeType const &hidden_nodes);
-  void LayerAddDenseActivation(fetch::vm::Ptr<fetch::vm::String> const &layer,
-                               math::SizeType const &inputs, math::SizeType const &hidden_nodes,
-                               fetch::vm::Ptr<fetch::vm::String> const &activation);
-
-  void LayerAddConv(fetch::vm::Ptr<fetch::vm::String> const &layer,
-                    math::SizeType const &output_channels, math::SizeType const &input_channels,
-                    math::SizeType const &kernel_size, math::SizeType const &stride_size);
-  void LayerAddConvActivation(fetch::vm::Ptr<fetch::vm::String> const &layer,
-                              math::SizeType const &                   output_channels,
-                              math::SizeType const &                   input_channels,
-                              math::SizeType const &kernel_size, math::SizeType const &stride_size,
-                              fetch::vm::Ptr<fetch::vm::String> const &activation);
+  template <typename... LayerArgs>
+  void AddLayer(fetch::vm::Ptr<fetch::vm::String> const &layer, LayerArgs... args);
+  template <typename... LayerArgs>
+  void AddLayerSpecificImpl(SupportedLayerType layer, SequentialModelPtr me,
+                            math::SizeType const &inputs, math::SizeType const &hidden_nodes);
+  template <typename... LayerArgs>
```

Comment:
not obvious why you need the parameter packs for the AddLayerSpecificImpl methods?

Reply:
oops, an artifact :)